src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/fluentd/FluentdWriter.java

```diff
@@ -0,0 +1,327 @@
+    }
+
+    public int getBufferRetentionTimeMillis() {
+        return bufferRetentionTimeMillis;
+    }
+
+    @DataBoundSetter
+    public void setBufferRetentionTimeMillis(int bufferRetentionTimeMillis) {
+        this.bufferRetentionTimeMillis = bufferRetentionTimeMillis;
+    }
+
+    public int getMaxRetries() {
+        return maxRetries;
+    }
+
+    @DataBoundSetter
+    public void setMaxRetries(int maxRetries) {
+        this.maxRetries = maxRetries;
+    }
+
+    public int getMaxWaitSeconds() {
+        return maxWaitSeconds;
+    }
+
+    @DataBoundSetter
+    public void setMaxWaitSeconds(int maxWaitSeconds) {
+        this.maxWaitSeconds = maxWaitSeconds;
+    }
+
+    public String getTag() {
+        return tag;
+    }
+
+    @DataBoundSetter
+    public void setTag(String tag) {
+        this.tag = tag;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    @DataBoundSetter
+    public void setHost(String host) {
+        this.host = host;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    @DataBoundSetter
+    public void setPort(int port) {
+        if(port < 0 || port > 65535) throw new IllegalArgumentException("Port out of range: " + port);
+        this.port = port;
+    }
+
+    public int getRetryMillis() {
+        return retryMillis;
+    }
+
+    @DataBoundSetter
+    public void setRetryMillis(int retryMillis) {
+        if (retryMillis < 0)
+            retryMillis = 0;
+        this.retryMillis = retryMillis;
+    }
+
+    public int getTimeoutMillis() {
+        return timeoutMillis;
+    }
+
+    @DataBoundSetter
+    public void setTimeoutMillis(int timeoutMillis) {
+        if(timeoutMillis < 0) throw new IllegalArgumentException("timeoutMillis less than 0");
+        this.timeoutMillis = timeoutMillis;
+    }
+
+    public int getBufferCapacity() {
+        return bufferCapacity;
+    }
+
+    @DataBoundSetter
+    public void setBufferCapacity(int bufferCapacity) {
+        if (bufferCapacity < 0)
+            bufferCapacity = DEFAULT_BUFFER_CAPACITY;
+        this.bufferCapacity = bufferCapacity;
+    }
+
+
+    @Extension
+    @Symbol("fluentd")
+    public static class DescriptorImpl extends ElasticSearchWriteAccessDescriptor {
+        @Override
+        public String getDisplayName() {
+            return "Fluentd Writer";
+        }
+    }
+
+    /**
+     * Logs the given object to Fluentd asynchronously
+     * Checks if there have been any data that couldn't be sent after several retries and then fails.
+     *
+     * @param data The data to post
+     * @throws IOException if something failed permanently
+     */
+    @Override
+    public void push(Map<String, Object> data) throws IOException {
+        if(fluentd == null) initFluentdLogger();
+        if (data.containsKey(ConsoleNotes.MESSAGE_KEY)) {
+            String message = (String) data.get(ConsoleNotes.MESSAGE_KEY);
+            if (message.length() > bufferCapacity * 2) {
```

Comment:
Why `*2`? Wouldn't `message.length() > bufferCapacity` be already too big?

Reply:
The lib has basically 3 buffer settings. The initial capacity, this is what we  refer to here. Then it has a chunk retention size. As soon as the buffer is filled above this limit data will be flushed. This is set to 3 times the initial capacity. Finally there is the maxbuffer size. This is set to 10 times the initial capacity. Worst case a string consists only of characters that need 4 bytes. So we should always be able to get it into the buffer. We also have the metadata in the events that also requires some space in the buffer.